sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1761,4 +1763,67 @@ private static void checkEmptyStringAllowed(Class<?> type, JavaType genericType,
     }
     return supportedRunners.build();
   }
+
+  /**
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties.
+   */
+  public static PipelineOptions fromProperties(final Properties properties) {
+    if (properties == null) {
+      return create();
+    }
+    final String[] args = properties.stringPropertyNames()
+      .stream()
+      .map(k -> String.format("--%s=%s", k, properties.getProperty(k)))
+      .toArray(String[]::new);
+    return fromArgs(args).create();
+  }
+
+  /**
+   * @param prefix the prefix filter applied on system properties keys.
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties
+   * filtered with the specified prefix.
+   */
+  public static PipelineOptions fromProperties(final String prefix, final Properties properties) {
+    if (prefix == null) {
+      return fromProperties(properties);
+    }
+    if (properties == null) {
+      return create();
+    }
+    final Properties instantiationProperties = properties.stringPropertyNames()
+      .stream()
+      .filter(k -> k.startsWith(prefix))
+      .collect(
+        Properties::new,
+        (p, k) -> p.setProperty(k.substring(prefix.length()), properties.getProperty(k)),
+        Hashtable::putAll);
+    return fromProperties(instantiationProperties);
+  }
+
+  /**
+   * @param prefix the prefix filter applied on system properties keys.
+   * @return a pipeline options instance based on current environment and overriden with
+   * system properties key/values and filtered with a custom prefix.
+   */
+  public static PipelineOptions fromJvm(final String prefix) {
```

Comment:
`fromSystemProperties`

Reply:
notr really it is the env overriden with the system properties, so it is really fromJvm. Strictly speaking it could be fromEnvironmentAndSystemProperties but it is way too long to be sane ;).

Reply:
Don't override it from the env :-)

Reply:
Sorry to ask another time Kenn but I dont get the rational, this is a very common strategy all config projects use, why would it break it?